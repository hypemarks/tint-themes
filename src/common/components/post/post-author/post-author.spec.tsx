import { render } from '@testing-library/react';
import React from 'react';
import PostAuthor from './post-author';
import { IAuthor } from '@tintup/tint-sdk/lib';

describe('Test <PostAuthor /> Component', () => {
  const author: IAuthor = { image_url: '', username: 'username', name: 'name', url: 'url' };

  it('should render snapshot', () => {
    const wrapper = render(<PostAuthor author={author} showName={true} showUsername={true} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should render author name without username', () => {
    const wrapper = render(<PostAuthor author={author} showName={true} showUsername={false} />);
    const authorName = wrapper.queryByTestId('author-name');
    const authorUserName = wrapper.queryByTestId('author-username');

    expect(authorName).toBeInTheDocument();
    expect(authorUserName).not.toBeInTheDocument();
  });

  it('should render user name without author', () => {
    const wrapper = render(<PostAuthor author={author} showName={false} showUsername={true} />);
    const authorName = wrapper.queryByTestId('author-name');
    const authorUserName = wrapper.queryByTestId('author-username');

    expect(authorName).not.toBeInTheDocument();
    expect(authorUserName).toBeInTheDocument();
  });
});
