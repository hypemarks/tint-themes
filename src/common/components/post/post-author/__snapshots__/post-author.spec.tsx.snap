// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PostAuthor /> Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="post-author"
  >
    <a
      aria-label="Posted by username"
      class="post-author__link"
      data-testid="link"
      href="url"
      rel="noopener noreferrer"
      tabindex="0"
      target="_blank"
    >
      <span
        class="post-author__author-name"
        data-testid="author-name"
        style="color: rgb(0, 0, 0);"
      >
        name
      </span>
      <span
        class="post-author__username"
        data-testid="author-username"
        style="color: rgb(204, 204, 204);"
      >
        @username
      </span>
    </a>
  </div>
</DocumentFragment>
`;
